test/testcafe/framework/page-objects/ChallengeCustomAppPushPageObject.js

```diff
@@ -29,51 +53,100 @@ export default class ChallengeCustomAppPushPageObject extends ChallengeFactorPag
   }
 
   getResendPushButton() {
+    if (userVariables.v3) {
+      return this.form.getButton('Resend push notification');
+    }
     return this.form.getElement('.button-primary');
   }
 
+  getResendPushButtonText() {
+    if (userVariables.v3) {
+      return this.getResendPushButton().textContent;
+    }
+    return this.getResendPushButton().value;
+  }
+
   clickResendPushButton() {
-    return this.form.clickSaveButton();
+    return this.form.clickButton('Resend push notification');
   }
 
   async waitForErrorBox() {
     await this.form.el.find(FORM_INFOBOX_ERROR).exists;
   }
 
   getErrorBox() {
-    return this.form.getElement(FORM_INFOBOX_ERROR);
+    return this.form.getErrorBox();
   }
 
   getErrorTitle() {
+    if (userVariables.v3) {
+      return this.getErrorBox();
+    }
     return this.form.getElement(FORM_INFOBOX_ERROR_TITLE);
   }
 
   getWarningBox() {
-    return this.form.getElement(FORM_INFOBOX_WARNING);
+    return this.form.getAlertBox();
   }
 
   async autoChallengeInputExists() {
+    if (userVariables.v3) {
+      return this.form.getCheckbox('Send push automatically').exists;
+    }
     return this.form.elementExist(AUTO_CHALLENGE_CHECKBOX_SELECTOR);
   }
 
   async autoChallengeInputIsVisible() {
+    if (userVariables.v3) {
+      return this.autoChallengeInputExists();
+    }
     return this.form.getElement(AUTO_CHALLENGE_CHECKBOX_SELECTOR).visible;
   }
 
+  getAutoChallengeCheckBox() {
+    if (userVariables.v3) {
+      return queryByLabelText('Send push automatically');
+    }
```

Comment:
I actually expect all of these that are wrapped in the v3 conditional to also work for Gen 2, but maybe you are just trying to preserve the Gen 2 page object logic?

Reply:
yeah I was trying to preserve the gen 2 logic, but youre right they do work in g2 so I think its safe to remove the g2 logic